### PR TITLE
Fix intervaltree min max

### PR DESCRIPTION
**Pull request by BenFrantzDale**

### CODE REVIEW

File: IntervalTree.h [hunk 1]
```diff
@@ -96,12 +96,15 @@ class IntervalTree {
       , right(nullptr)
     {
         --depth;
-        const auto minmaxStop = std::minmax_element(ivals.begin(), ivals.end(), 
-                                                    IntervalStopCmp());
-        const auto minmaxStart = std::minmax_element(ivals.begin(), ivals.end(), 
-                                                     IntervalStartCmp());
-        if (!ivals.empty()) {
-            center = (minmaxStart.first->start + minmaxStop.second->stop) / 2;
+        {   // This scope is because afterward we sort and so these two iterators
```
Comment (BenFrantzDale) [inline IntervalTree.h:99]: This is just scoping

File: IntervalTree.h [hunk 2]
```diff
@@ -160,15 +164,27 @@ class IntervalTree {
 
     Scalar min() const {
         assert(!empty());
```
Comment (BenFrantzDale) [inline IntervalTree.h:166]: These are the fixes.

File: interval_tree_test.cpp [hunk 3]
```diff
@@ -186,3 +186,43 @@ int main(int argc, char**argv) {
 
     return Catch::Session().run( argc, argv );
 }
+
+TEST_CASE( "Test min/max" ) {
```
Comment (BenFrantzDale) [inline interval_tree_test.cpp:190]: This tests the fixes.

**Review by will-formlabs — APPROVED** [review]
Looks good

**Review by dbiswas2808 — APPROVED** [review]
LGTM. Just a thought, is it a good idea to just a cache the min and max when we are building the tree for each level?

**Pull request merged**